### PR TITLE
feat: Java workload e2e

### PR DESCRIPTION
**Pull request by yodigos**

### CODE REVIEW

File: tests/e2e/workload-lifecycle/services/java-http-server/java-supported-version.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM maven:3.8.5-openjdk-17 AS build
+COPY src /home/app/src
+COPY pom.xml /home/app
+RUN mvn -f /home/app/pom.xml clean package
+
+FROM eclipse-temurin:17-jre-jammy
+COPY --from=build /home/app/target/*.jar /app/java-supported-version.jar
+USER 15000
+CMD ["java","-jar", "/app/java-supported-version.jar"]
```
Comment (blumamir) [inline tests/e2e/workload-lifecycle/services/java-http-server/java-supported-version.Dockerfile:9]: Are there any tests for unsupported version?

File: tests/e2e/workload-lifecycle/services/java-http-server/java-supported-docker-env.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FROM maven:3.8.5-openjdk-17 AS build
+COPY src /home/app/src
+COPY pom.xml /home/app
+RUN mvn -f /home/app/pom.xml clean package
+
+FROM eclipse-temurin:17-jre-jammy
+ENV JAVA_OPTS="-Dthis.does.not.exist=true"
```
Comment (blumamir) [inline tests/e2e/workload-lifecycle/services/java-http-server/java-supported-docker-env.Dockerfile:7]: Where do we assert that this value indeed exists in the application?
If odigos remove it when env overwriting, will the test fail?

**Review by blumamir — COMMENTED** [review]
Please write a description about the scenarios [here](https://github.com/odigos-io/odigos/tree/main/tests/e2e/workload-lifecycle).

I see we don't have any tests for unsupported or old versions. Since this is where many problems are manifested, I think it even more important to test these then the mainstream version.

Also, please add a test for java latests version, as if something breaks with new versions, we will not have any other indication. Should we also test the minimum version?

**Review by blumamir — APPROVED** [review]
